server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataMappingService.java

```diff
@@ -239,6 +241,21 @@ public void refreshMapping(final String index, final String indexUUID) {
             }
         }
 
+        private DocumentMapper getMapperForUpdate(MapperService mapperService, String type) {
+            DocumentMapper mapper = mapperService.documentMapper(type);
+            if (mapper == null && type.equals(MapperService.SINGLE_MAPPING_NAME) &&
```

Comment:
Instead of checking the index version here, would it make sense to check if the index has multiple mappings? That would be consistent with how typeless 'get mappings' works on indices with multiple types.

Reply:
Another thing to have in mind is that 5.x indices can't be read by Elasticsearch 7.x anyway due to our backward compatibility policy that only supports reading indices created by version N or N-1. So 5.x indices would have to be reindexed prior to upgrading anyway.

Reply:
Okay, this reasoning makes sense to me as well.